hm_pyhelper/miner_json_rpc/client.py

```diff
@@ -58,3 +61,55 @@ def get_peer_book(self):
     def get_firmware_version(self):
         summary = self.get_summary()
         return summary.get('firmware_version')
+
+    def create_add_gateway_txn(self, gateway_address: str,
+                               owner_address: str, payer_address: str) -> dict:
+        """
+        Invokes the txn_add_gateway RPC endpoint on the miner and returns
+        the same payload that the smartphone app traditionally expects.
+        https://docs.helium.com/mine-hnt/full-hotspots/become-a-maker/hotspot-integration-testing/#generate-an-add-hotspot-transaction
+
+        Alternatively, the same thing can be accomplished with dbus,
+        like in the below, but RPC is generally easier to use.
+        https://github.com/NebraLtd/hm-config/blob/900aeed353fb9729b49bca97d7da8a9abf0a2029/gatewayconfig/bluetooth/characteristics/add_gateway_characteristic.py#L71
+
+        There are a few limitations to keep in mind when generating these
+        transactions:
+        1. None of the addresses returned can be properly deserialized.
```

Comment:
Guess we should report these issues to helium if we haven't already?

Reply:
I will update these comments. Ended up figuring out that the missing 0 is for version which erlang miner strips out and the app adds back in.

Reply:
The missing payer part in #2 is an open question and I'll raise that issue.

Reply:
Actually with my recent findings around version, I'm going to redo a bit of this. Converting to draft.